cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/documentstore/AttachmentManager.java

```diff
@@ -238,12 +239,12 @@ protected static PreparedAttachment prepareAttachment(String attachmentsDir,
      * @param attachments Attachments to search.
      * @return List of attachments which already exist in the attachment store, or an empty list if none.
      */
-    public static List<SavedAttachment> findExistingAttachments(
-            Collection<? extends Attachment> attachments) {
-        ArrayList<SavedAttachment> list = new ArrayList<SavedAttachment>();
-        for (Attachment a : attachments) {
+    public static Map<String, SavedAttachment> findExistingAttachments(
+            Map<String, ? extends Attachment> attachments) {
+        Map<String, SavedAttachment> list = new HashMap<String, SavedAttachment>();
```

Comment:
The field name `list` is a bit confusing now, since this isn't a `List` any more. I'm not massively concerned about this because the method is concise and so it is clear what is going on, but it might be better named `existingAttachments` or `attachments`.